IPython/utils/module_paths.py

```diff
@@ -64,20 +56,42 @@ def find_module(name, path=None):
         Return full path of module or None if module is missing or does not have
         .py or .pyw extension
     """
-    if name is None:
-        return None
-    try:
-        file, filename, _ = imp.find_module(name, path)
-    except ImportError:
-        return None
-    if file is None:
-        return filename
-    else:
-        file.close()
-    if os.path.splitext(filename)[1] in [".py", ".pyc"]:
-        return filename
-    else:
-        return None
+    return _find_module(name, path)
+
+
+if sys.version_info >= (3,4):
+    import importlib.util
+    def _find_module(name, path):
+        if path is not None:
+            warnings.warn( "`path=` argument deprecated on python 3.4 and above", DeprecationWarning, stacklevel=2)
```

Comment:
I think this should be more explicit - it's not just deprecated, it's actually ignored.